application/src/main/java/org/togetherjava/tjbot/features/moderation/history/MessageHistoryRoutine.java

```diff
@@ -0,0 +1,64 @@
+package org.togetherjava.tjbot.features.moderation.history;
+
+import net.dv8tion.jda.api.JDA;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.MessageHistoryRecord;
+import org.togetherjava.tjbot.features.Routine;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static org.togetherjava.tjbot.db.generated.Tables.MESSAGE_HISTORY;
+
+/**
+ * Routine that deletes records from message_history post expiration hours.
+ */
+public final class MessageHistoryRoutine implements Routine {
```

Comment:
It cleans up records from `DB` that are older than our purge limit aka 24 hrs, i can't think of better names but do agree it needs to be changed. 
I asked gpt for some help, what you think suits best?
![image](https://github.com/Together-Java/TJ-Bot/assets/61616007/92502232-a825-4fac-b84b-94ad7d3fb0a7)

Reply:
Depends on the reason we are deleting them. If we are deleting them with hatred, maybe Eradicatate prefix, since that's quite aggressive word. Synonms are: abolish, destroy, obliterate, exterminate.. 

If we are deleting them because they are dirty, maybe cleanse, sanitize, CleansingRitual.. or my favourite, that is not on the list, purify. Because purify means to get rid of impurities, to make something pure. But pure has so many different meanings, and it's quite beatuiful word, that is seen everywhere, from sanskrit, old french, old high german..

> from Latin purus "clean, clear; unmixed; unadorned; chaste, undefiled."
> 
> This is conjectured to be from PIE root *peue- "to purify, cleanse" (source also of Latin putus "clear, pure;"

Other gpt suggested terms are computer science related:
- Flush is often used in context of buffers, and basically means empty the buffer. (real world association to water and toilet flushing)
- Prune is often used in context of data structures and algorithms, for example, you might prune a tree, which means remove some branches, to reduce complexity. (real world association is tree pruning, literal removal of branches)
- Purge is often used in context of cache invalidation, and means remove cache before it would naturally expire, or be evicted. Purges are usually ran periodically, based on some criteria. On databases it usually means deleting obsolate data, that is no longer needed. (real world association to getting rid of something unwanted)

Hope that helps :relaxed:

Reply:
thank you for such an insightful explanation 😀, i think Purge suits this one.